--- ./driver/Makefile.in.orig	2021-04-01 13:16:58.000000000 +0300
+++ ./driver/Makefile.in	    2021-04-24 16:59:04.705436514 +0300
@@ -212,7 +212,7 @@
 		  xdpyinfo      test-screens    test-yarandom   test-xinput   \
 	          test-xkb
 
-EXES		= xscreensaver xscreensaver-command xscreensaver-settings
+EXES		= xscreensaver xscreensaver-command
 UTIL_EXES	= xscreensaver-gfx @EXES_SYSTEMD@
 SETUID_EXES	= xscreensaver-auth
 DEMO_EXES	= @ALL_DEMO_PROGRAMS@
@@ -222,7 +222,7 @@
 		  xscreensaver.h prefs.h remote.h exec.h \
 		  demo-Gtk-conf.h auth.h types.h blurb.h atoms.h clientmsg.h \
 		  screens.h xinput.h fade.h
-MENA		= xscreensaver.man xscreensaver-settings.man \
+MENA		= xscreensaver.man \
 		  xscreensaver-command.man
 MENB		= xscreensaver-gfx.man xscreensaver-auth.man \
 		  xscreensaver-command.man xscreensaver-systemd.man
@@ -306,11 +306,8 @@
 	 done
 
 
-# Symlink from xscreensaver-demo -> xscreensaver-settings in /usr/bin/
 install-program::
-	@D=$(install_prefix)$(bindir) ;					      \
-	 echo ln -sf xscreensaver-settings $$D/xscreensaver-demo ; 	      \
-	      ln -sf xscreensaver-settings $$D/xscreensaver-demo
+	@D=$(install_prefix)$(bindir)					      \
 
 install-ad: XScreenSaver.ad
 	@if [ ! -d $(install_prefix)$(AD_DIR) ]; then			      \
@@ -384,7 +381,6 @@
 # Symlink from xscreensaver-demo.1 -> xscreensaver-settings.1 in /usr/man/man1/
 install-man::
 	@D=$(install_prefix)$(mandir)/man$(mansuffixA) ;		\
-	 F1=xscreensaver-settings.$(mansuffixA) ;			\
 	 F2=xscreensaver-demo.$(mansuffixA) ;				\
 	 echo ln -sf $$F1 $$D/$$F2 ;					\
 	      ln -sf $$F1 $$D/$$F2
